src/cmap/connection_pool.ts

```diff
@@ -640,18 +653,19 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
           this[kConnections].push(connection);
```

Comment:
if pool is closed or paused, destroy the connection, don't add to `[kConnections]`